internal/certprovider_test.go

```diff
@@ -37,21 +128,101 @@ func TestNitroCertProvider_Roots(t *testing.T) {
 		assert.False(t, gotRoots.Equal(x509.NewCertPool())) // check not empty
 	})
 
-	errorTests := []struct {
+	t.Run("happy path - multiple calls", func(t *testing.T) {
+		// We need to test multiple calls to Roots(). because the first call
+		// closes rootCertZipReader after which Roots() returns cached values.
+
+		// given
+		cp := internal.NewNitroCertProvider(
+			internal.NewEmbeddedRootCertZipReader(),
+		)
+
+		// when
+		_, err := cp.Roots()
+		require.NoError(t, err)
+		_, err = cp.Roots()
+
+		// then
+		require.NoError(t, err)
```

Comment:
I don't know how much it really matters in this case, but feels like we should still assert roots in the same way that we do in the first "happy path" test case instead of ignoring them.
```golang
// when - make sure the first call still returns "correct" roots
gotRoots, err := cp.Roots()
require.NoError(t, err)
require.False(t, gotRoots.Equal(x509.NewCertPool())) // check not empty

gotRoots, err = cp.Roots()

// then - now we test that subsequent calls also return "correct" roots
require.NoError(t, err)
assert.False(t, gotRoots.Equal(x509.NewCertPool())) // check not empty
```

Reply:
I'll check the return value of the second call

Reply:
done